libchihua-msg/src/orchestrator/controller.c

```diff
@@ -91,7 +92,10 @@ void controller_unrefp(Controller **controller) {
 }
 
 // NOLINTNEXTLINE
-static int accept_handler(sd_event_source *s, int fd, uint32_t revents, void *userdata) {
+static int accept_handler(sd_event_source *source, int fd, uint32_t revents, void *userdata) {
+        NOT_USED(source);
```

Comment:
I am not sure about the `NOT_USED`. 
Unused parameters usually detected via linters in other languages - and in our case we have the [clang-tidys misc-unused-parameters check](https://clang.llvm.org/extra/clang-tidy/checks/misc/unused-parameters.html). In this case, the `clang-tidy` error has been explicitly silenced via the `// NOLINTNEXTLINE` since the function is still "under construction". 
The `NOT_USED` is extra code just to silence the compiler. Where instead the `// NOLINTNEXTLINE` is explicitly for the static analysis. So I'd prefer the linting to detect it - which would also mean that compiler warnings can't be errors.

Reply:
This rule of not having unused function arguments is akin to removing dead code, which is a good idea for having more readable, more maintainable code.  Sometimes unused function arguments can't be removed.  This solution with NOT_USED makes the compiler happy, but it also makes the function more readable, because you can see at a glance whether certain arguments will be used in the function.

The extra code is very minimal, because it's just a cast.  It might even be optimized away by the compiler.

I suggest we remove the ```NOLINTNEXTLINE``` above this function.  That said, I like the ```NOLINTNEXTLINE``` and will look for opportunities to use it going forward.

Reply:
The `NOLINTNEXTLINE` is really just a fallback and should only be used for good reasons. 

I totally agree that unused arguments should be removed in order to get readable and concise code. Other languages are using an underscore `_` for this, which is much more concise and has the big advantage of requiring to define the argument name as soon as you use it. Thats something I am missing with this approach - you have to actively remove it. But I guess this is just C-style again. 
However, can we change the cast to a definition of the `unused` attribute? E.g. like this:
```C
#define UNUSED __attribute__((unused))
// ...
static int accept_handler(UNUSED sd_event_source *s, UNUSED int fd, UNUSED uint32_t revents, UNUSED void *userdata) {
```
In that case, please also remove the `// NOLINTNEXTLINE`.

Reply:
Very cool!  I wasn't aware of this GCC attribute.  This is a much better solution than the cast.